rfcs/convergence/authoring-stories.md

```diff
@@ -267,13 +271,144 @@ TBA
 
 TBA
 
-### 7. how structure stories in the storybook nav tree
+### 7. dissecting big story files into smaller ones
 
-TBA
+Components with complex API require many long stories, both for documentation and for e2e testing. This creates big story files, which are hard to maintain. How can these files be dissected into smaller ones?
 
-### 8. dissecting big story files into smaller ones
+#### Proposal
 
-TBA
+1. Every component can have at most one `.stories.tsx` file with default export which configures metadata about the component. This file must be called `Component.stories.tsx`, for example `Button.stories.tsx`.
+2. If putting all stories into one file would make it too long, individual stories might be put into additional `.stories.tsx` files as a named export, and then re-exported from `Component.stories.tsx` file like this: `export * from ‘./IndividualStoryFile.stories’;`
+3. If individual story files are employed, `Component.stories.tsx` file must not contain any stories besides the default export.
+
+
+##### Good Example 1 - single file
+
+Button.stories.tsx:
+```tsx
+import { Meta } from '@storybook/react';
+
+export const Default = (props: ButtonProps) => <Button {...props}>Button</Button>;
+export const ButtonWithIcon = () => <Button icon={<CalendarIcon />}>Text</Button>;
+
+export default {
+  title: 'Components/Button',
+  component: Button
+} as Meta;
+```
+
+##### Good Example 2 - multiple files
+
+Button.stories.tsx:
+```tsx
+import { Meta } from '@storybook/react';
+
+export * from "ButtonDefault.stories"
+export * from "ButtonWithIcon.stories"
+
+export default {
+  title: 'Components/Button',
+  component: Button
+} as Meta;
+```
+
+ButtonDefault.stories.tsx:
+```tsx
+export const ButtonDefault = (props: ButtonProps) => <Button {...props}>Button</Button>;
+ButtonDefault.storyName = "Default"
+```
+
+ButtonWithIcon.stories.tsx:
+```tsx
+export const ButtonWithIcon = () => <Button icon={<CalendarIcon />}>Text</Button>;
+```
+
+##### Bad example - multiple files
+
```

Comment:
can we add 2 examples that cover all the donts?

- 1. dont define CFS within your stories if you are using "multiple files" pattern
- 2. dont mix  re-exports with inline definition within main story ( story that will be rendered in tree )

Reply:
The first point is already addressed by the first example, right? I’ve added a second example to address the second point :).